File: src/com/kedzie/vbox/machine/settings/SystemFragment.java
package com.kedzie.vbox.machine.settings;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.actionbarsherlock.app.SherlockFragment;
import com.kedzie.vbox.R;
import com.kedzie.vbox.app.FragmentElement;
import com.kedzie.vbox.app.PagerTabHost;

public class SystemFragment extends SherlockFragment {
    private PagerTabHost mTabHost;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        mTabHost = new PagerTabHost(getActivity());
        mTabHost.setup(getActivity(), getChildFragmentManager(), R.id.realtabcontent);
        mTabHost.addTab(new FragmentElement("Motherboard", SystemMotherboardFragment.class, getArguments()));
        mTabHost.addTab(new FragmentElement("Processors", SystemProcessorsFragment.class, getArguments()));
        mTabHost.addTab(new FragmentElement("Accleration", SystemAccelerationFragment.class, getArguments()));
        
//        mTabHost.addTab(mTabHost.newTabSpec("motherboard").setIndicator("Motherboard"), SystemMotherboardFragment.class, getArguments());
//        mTabHost.addTab(mTabHost.newTabSpec("processors").setIndicator("Processors"), SystemProcessorsFragment.class, getArguments());
//        mTabHost.addTab(mTabHost.newTabSpec("acceleration").setIndicator("Acceleration"), SystemAccelerationFragment.class, getArguments());
        return mTabHost;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mTabHost = null;
    }
}

File: vbox-app/src/main/java/com/kedzie/vbox/machine/settings/SystemFragment.kt
package com.kedzie.vbox.machine.settings


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentPagerAdapter
import com.kedzie.vbox.R
import kotlinx.android.synthetic.main.pager_tabs.*

class SystemFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.pager_tabs, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        pager.adapter = object : FragmentPagerAdapter(childFragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
            override fun getCount() = 3

            override fun getItem(position: Int): Fragment {
                return when (position) {
                    0 -> SystemMotherboardFragment(arguments!!)
                    1 -> SystemProcessorsFragment(arguments!!)
                    else -> SystemAccelerationFragment(arguments!!)
                }
            }

            override fun getPageTitle(position: Int): CharSequence {
                return when (position) {
                    0 -> "Motherboard"
                    1 -> "Processors"
                    else -> "Acceleration"
                }
            }
        }
        pager_tabs.setupWithViewPager(pager)
    }
}

